modules/core/src/test/java/org/apache/synapse/config/xml/HeaderMediatorSerializationTest.java

```diff
@@ -34,15 +34,30 @@ public HeaderMediatorSerializationTest() {
         headerMediatorSerializer = new HeaderMediatorSerializer();
     }
 
-    public void testHeaderMediatorSerializationSenarioOne() throws Exception {
-        String inputXml = "<header xmlns=\"http://ws.apache.org/ns/synapse\" name=\"To\" value=\"http://127.0.0.1:10001/services/Services\"/>";
+    public void testHeaderMediatorSerializationScenarioOne() throws Exception {
+        String inputXml = "<header xmlns=\"http://ws.apache.org/ns/synapse\" name=\"To\" " +
+                "value=\"http://127.0.0.1:10001/services/Services\"/>";
         assertTrue(serialization(inputXml, headerMediatorFactory, headerMediatorSerializer));
         assertTrue(serialization(inputXml, headerMediatorSerializer));
     }
 
-    public void testHeaderMediatorSerializationSenarioTwo() throws Exception {
+    public void testHeaderMediatorSerializationScenarioTwo() throws Exception {
         String inputXml = "<header xmlns=\"http://ws.apache.org/ns/synapse\" name=\"To\" action=\"remove\"/>";
         assertTrue(serialization(inputXml, headerMediatorFactory, headerMediatorSerializer));
         assertTrue(serialization(inputXml, headerMediatorSerializer));
     }
+
+    public void testHeaderMediatorSerializationScenarioThree() throws Exception {
+        String inputXml = "<header xmlns=\"http://ws.apache.org/ns/synapse\" name=\"To\" " +
+                "expression=\"fn:concat('http://localhost:9764/services/Axis2SampleService_',get-property('epr'))\"/>";
+        assertTrue(serialization(inputXml, headerMediatorFactory, headerMediatorSerializer));
+        assertTrue(serialization(inputXml, headerMediatorSerializer));
+    }
+
+    public void testHeaderMediatorSerializationScenarioFour() throws Exception {
```

Comment:
Rather than naming these as scenario1, scenario2. scenario3... it's better if we can have sensible names meaning what the test is about. Otherwise, we'll have to look into the code to inderstand what it tests.

Reply:
Fixed in https://github.com/wso2/wso2-synapse/pull/1445